koku/api/query_handler.py

```diff
@@ -298,3 +298,44 @@ def _get_filter(self, delta=False):
         filters.add(query_filter=end_filter)
 
         return filters
+
+    def filter_to_order_by(self, parameters):
+        """Remove group_by[NAME]=* and replace it with group_by[NAME]=X.
+
+        The parameters object contains a list of filters and a list of group_bys.
+
+        For example, if the parameters object contained the following:
+        group_by[X] = Y
+        group_by[Z] = *     # removes this line
+        filter[Z] = L
+        filter[X] = Y
+
+        The returned parameters object would contain lists that look like this:
+
+        group_by[X] = Y
+        group_by[Z] = L     # adds this line
+        filter[Z] = L
+        filter[X] = Y
+
+        Thereby removing the star when there is a filter provided.
+
+        Args:
+            parameters (QueryParameters): The parameters object
+
+        Returns:
+            parameters (QueryParameters): The parameters object
+
+        """
+        for group_by_key in parameters.parameters['group_by']:
+            for group_by_value in parameters.parameters['group_by'][group_by_key]:
+                if group_by_value == '*':
+                    # find if there is a filter[X]=Y that matches this group_by[X]=*
+                    for filter_key in parameters.parameters['filter']:
```

Comment:
Why do we need to loop through all of the filters and later check if the `filter_key` and `group_by_key` are equal?

Wouldn't it make more sense to check if the `group_by_key` was in the filters up front?
```
filter_value = parameters.parameters.get('filter', {}).get('group_by_key')
if filter_value:
    parameters.parameters['group_by'][group_by_key] = filter_value
```

Reply:
@chambridge Is your concern that you want to optimize in order to make this faster?

Reply:
Yes, this is a primary code path so I would like to avoid unnecessary cpu cycles.

Reply:
Can we optimize later? I’d like Luke to start writing tests while I optimize further.

Reply:
I would suggest tackling what seems like an unnecessary loop no based on the above. 
The major focus of the current sprint is query performance improvement so adding extra time seems like the wrong choice at the moment.

Luke should still be able to write tests, the would just be expected to fail until this change goes in.

Reply:
@chambridge what is your time complexity goal for this function?

Reply:
You just need to avoid looping over all filter keys above. A loop is not necessary as you only need to proceed down the code path if the `group_by_key` is present in the filters.

Currently if you had
`?group_by[region]=*&filter[service]=EC2&filter[account]=1234`
Your code would loop through all the filters even though there is no work to be done.

In the next case:
`?group_by[region]=*&filter[service]=EC2&filter[account]=1234&filter[region]=us-east-1`
Your code would loop through 3 filters when only one is applicable, so can just look up the filter using the group_by key `region`.

Reply:
I optimized this loop now.